library/core/src/slice/ascii.rs

```diff
@@ -80,6 +80,84 @@ impl [u8] {
     pub fn escape_ascii(&self) -> EscapeAscii<'_> {
         EscapeAscii { inner: self.iter().flat_map(EscapeByte) }
     }
+
+    /// Returns a byte slice with leading ASCII whitespace removed.
+    ///
+    /// 'Whitespace' refers to the definition used by
+    /// [`u8::is_ascii_whitespace`].
+    ///
+    /// [`u8::is_ascii_whitespace`]: primitive.u8.html#method.is_ascii_whitespace
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(byte_slice_trim_ascii)]
+    ///
+    /// assert_eq!(b" \t hello world\n".trim_start_ascii(), b"hello world\n");
+    /// assert_eq!(b" ".trim_start_ascii(), b"");
+    /// ```
+    #[unstable(feature = "byte_slice_trim_ascii", issue = "none")]
+    pub fn trim_start_ascii(&self) -> &[u8] {
```

Comment:
Not sure if this could be made `const`. Right now it doesn't work because `for` cannot be used in const functions, but maybe there's a workaround for that.

Reply:
If you want to make it const, you can replace the for loop with a similar while loop + a TODO to put back the for once it's supported in const functions.

Reply:
Thanks, I wasn't aware that `while` works in a const context. Actually the solution with `while` is more elegant than with the for loop, because the index can be re-used. However, I couldn't find a way to create a subslice, since both `self.get_unchecked(...)` and `&self[...]` are not const.

Reply:
Why aren't get_unchecked/get_unchecked_mut const? Perhaps they can be. So perhaps you can add const to them in a different commit...

Reply:
Not easily. For example, `self.get_unchecked` defers to `SliceIndex::get_unchecked`, which is a trait method that cannot be made const right now.